.github/workflows/deploy.yml

```diff
@@ -0,0 +1,53 @@
+name: Deploy
+
+on:
+  push:
+    branches:
+      - develop
+      - master
+
+jobs:
+  canary:
+    runs-on: ubuntu-latest
+    if: github.ref_name == 'develop'
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - uses: actions/cache@v3
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
+      # Make sure the working tree is clean
+      - run: git reset --hard HEAD
```

Comment:
Was this here previously for some reason that was specific to circleci? This feels like something that should be a "fail the build if not true" step, not a "silently force it to be true" step